### ISSUE TITLE
[question] Run test package dependently 

### ISSUE THREAD

**Issue by xakod** [conan-io/conan#15072]
### What is your question?

Hi! Can I run test package only if package will build with some options? For example
```python
    def test(self):
        if can_run(self):
            if self.dependencies[self.tested_reference_str].options.with_myoption:
                bin_path = os.path.join(
                    self.cpp.build.bindirs[0], 'PackageTest',
                )
                self.run(bin_path, env='conanrun')

            
```

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
It would be recommended in general to use the ``generate()`` method to generate the necessary information for the later "build" and "test" methods, and these methods should act accordingly. 
But I guess yes, for this specific use case, the ``dependencies`` field is available in Conan 2.0, so you can check it to get information from your dependencies, yes.

**Comment by xakod** [comment]
Thanx!
